tests/integration/tests.py

```diff
@@ -48,36 +51,376 @@ def launch(self, env_vars=None):
             check=True)
 
 
-def test_gpt_neo():
-    with Runner('deepspeed') as r:
-        prepare.build_hf_handler_model("gpt-neo-2.7b")
-        r.launch()
-        client.run("huggingface gpt-neo-2.7b".split())
+class TestHfHandler:
+    # Runs on g5.12xl
+    def test_gpt_neo(self):
+        with Runner('lmi', 'test_gpt4all_lora') as r:
+            prepare.build_hf_handler_model("gpt-neo-2.7b")
+            r.launch()
+            client.run("huggingface gpt-neo-2.7b".split())
+
+    def test_bloom_7b(self):
+        with Runner('lmi', 'bloom-7b1') as r:
+            prepare.build_hf_handler_model("bloom-7b1")
+            r.launch()
+            client.run("huggingface bloom-7b1".split())
+
+    def test_llama2_7b(self):
+        with Runner('lmi', 'llama-2-7b') as r:
+            prepare.build_hf_handler_model("llama-2-7b")
+            r.launch()
+            client.run("huggingface llama-2-7b".split())
+
+    def test_gptj_6B(self):
+        with Runner('lmi', 'gpt-j-6b') as r:
+            prepare.build_hf_handler_model("gpt-j-6b")
+            r.launch()
+            client.run("huggingface gpt-j-6b".split())
+
+    def test_gpt4all_lora(self):
+        with Runner('lmi', 'gpt4all-lora') as r:
+            prepare.build_hf_handler_model("gpt4all-lora")
+            r.launch()
+            client.run("huggingface gpt4all-lora".split())
+
+    def test_streaming_bigscience_bloom_3b(self):
+        with Runner('lmi', 'bigscience/bloom-3b') as r:
+            prepare.build_hf_handler_model("bigscience/bloom-3b")
+            r.launch("CUDA_VISIBLE_DEVICES=1,2")
+            client.run("huggingface bigscience/bloom-3b".split())
+
+    def test_streaming_t5_large(self):
+        with Runner('lmi', 't5-large') as r:
+            prepare.build_hf_handler_model("t5-large")
+            r.launch("CUDA_VISIBLE_DEVICES=1")
+            client.run("huggingface t5-large".split())
+
+
+class TestTrtLlmHandler1:
+    # Runs on g5.12xl
+    def test_llama2_13b_tp4(self):
+        with Runner('tensorrt-llm', 'llama2-13b') as r:
+            prepare.build_trtllm_handler_model("llama2-13b")
+            r.launch("CUDA_VISIBLE_DEVICES=0,1,2,3")
+            client.run("trtllm llama2-13b".split())
```

Comment:
So I can better understand this section. We are adding env's to the tensorrt-llm container launch for CUDA device visibility, which makes sense when we limit the number of GPUs, but here we are attaching all GPUs -- is this necessary? If it is we may want to address the launch containers afterwards, unless there is something else that I am missing when we use trtllm.

Reply:
I kept this to maintain parity with the previous version, so I didn't spend a lot of time thinking about it. It may not be necessary but I think that should be a separate PR/discussion